GalaxyZooTouchTable/GalaxyZooTouchTable/Models/TableUserFactory.cs

```diff
@@ -17,18 +17,19 @@ enum UserType
 
     public class TableUser : INotifyPropertyChanged
     {
+        public virtual string Name { get; }
```

Comment:
I like that you are using virtual for this property, but why aren't you using it for the other properties in this class?

Reply:
Why does it need to be overridden? It seems comparable to `ThemeColor`, `StartButton` and `Avatar`

Reply:
Without the override, `Name` takes the `null` value of the base class (`TableUser`) and will not work with the messenger. Take a look at the use of the `Messenger` in this PR. The name property is used there to create unique values for the messenger dictionary.

Reply:
But you only ever make TableUser (inheriting) objects through the factory, right? So as long as  `Name` is defined in the object produced by the factory method then I don't see how `Name` would ever be null. How is `name` used any differently than `ThemeColor`?

Reply:
`ThemeColor` is referenced through bindings in views, so the property is correctly retrieved from the object (whether `FaceUser` or `PersonUser`). However, take a look at the `NotificationsViewModel`. That model only knows that the User is a `TableUser`, not the type. So it still looks at the base class for definitions.

Reply:
Huh, maybe I'm just confused by how C# sharp works. It may be worth making the other props in `TableUser` virtual as well just in case you may have to use those values in code in the future. Also, I think it makes it more explicit that those values aren't supposed to be defined on the `TableUser` and that `TableUser` should really just be a virtual class.

Reply:
In that case, I should probably revert back to marking `TableUser` as an abstract class, which is more aligned with what I want.  I'll push up a fix momentarily.